File: networking/src/main/java/net/remmintan/mods/minefortress/networking/c2s/C2SDestroyBuilding.java
package net.remmintan.mods.minefortress.networking.c2s;

import net.minecraft.network.PacketByteBuf;
import net.minecraft.server.MinecraftServer;
import net.minecraft.server.network.ServerPlayerEntity;
import net.remmintan.mods.minefortress.networking.interfaces.FortressC2SPacket;

import java.util.UUID;

public class C2SDestroyBuilding implements FortressC2SPacket {

    public final static String CHANNEL = "destroy_building";

    private final UUID buildingId;

    public C2SDestroyBuilding(UUID buildingId) {
        this.buildingId = buildingId;
    }

    public C2SDestroyBuilding(PacketByteBuf buf) {
        this.buildingId = buf.readUuid();
    }


    @Override
    public void handle(MinecraftServer server, ServerPlayerEntity player) {
        getManagersProvider(server, player).getBuildingsManager().destroyBuilding(buildingId);
    }

    @Override
    public void write(PacketByteBuf buf) {
        buf.writeUuid(buildingId);
    }
}

File: src/networking/java/net/remmintan/mods/minefortress/networking/c2s/C2SDestroyBuilding.kt
package net.remmintan.mods.minefortress.networking.c2s

import net.minecraft.network.PacketByteBuf
import net.minecraft.server.MinecraftServer
import net.minecraft.server.network.ServerPlayerEntity
import net.minecraft.util.math.BlockPos
import net.remmintan.mods.minefortress.core.interfaces.networking.FortressC2SPacket

class C2SDestroyBuilding(private val pos: BlockPos) : FortressC2SPacket {

    constructor(buf: PacketByteBuf) : this(BlockPos.fromLong(buf.readLong()))

    override fun handle(server: MinecraftServer, player: ServerPlayerEntity) {
        getManagersProvider(server, player).buildingsManager.destroyBuilding(pos)
    }

    override fun write(buf: PacketByteBuf) {
        buf.writeLong(pos.asLong())
    }

    companion object {
        const val CHANNEL: String = "destroy_building"
    }
}
